compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -550,30 +550,23 @@ fn collate_raw_dylibs(used_libraries: &[NativeLib]) -> Vec<(String, Vec<DllImpor
             } else {
                 format!("{}.dll", name)
             };
-            dylib_table
-                .entry(name)
-                .or_default()
-                .extend(lib.dll_imports.iter().map(|import| import.name));
+            dylib_table.entry(name).or_default().extend(lib.dll_imports.iter().cloned());
         }
     }
 
-    // FIXME: when we add support for ordinals, fix this to propagate ordinals.  Also figure out
-    // what we should do if we have two DllImport values with the same name but different
-    // ordinals.
+    // Rustc already signals an error if we have two imports with the same name but different
+    // calling conventions (or function signatures), so we don't have pay attention to those
+    // when ordering.
```

Comment:
Is there a test for this case?

Reply:
If both definitions are in the same scope, it results in E0428 ("the name `mumble` is defined multiple times").  I wasn't able to find a test for the case where the definitions have different calling conventions, but [test/ui/issues/issue-28472.rs](https://github.com/rust-lang/rust/blob/master/src/test/ui/issues/issue-28472.rs) comes close; it's two `extern "C"` functions with the same name and signature.

If the definitions are in different blocks (say, one inside a function and one at module top-level), then rustc just issues a lint warning: "`mumble` redeclared with a different signature;" this is covered by [test/ui/lint/clashing-extern-function.rs](https://github.com/rust-lang/rust/blob/master/src/test/ui/lint/clashing-extern-fn.rs).  Do we need to upgrade this to an error?

Reply:
I think we shouldn't touch the lint, but report a new hard error here if mismatching declarations are used with raw-dylib feature specifically.